### PR TITLE
ref(staff): Switch from staff feature flag to option

### PR DESCRIPTION
**Pull request by schew2381**
Closed in favor of https://github.com/getsentry/sentry/pull/67852

### CODE REVIEW

File: src/sentry/auth/staff.py [hunk 1]
```diff
@@ -51,6 +52,16 @@ def is_active_staff(request: HttpRequest | Request) -> bool:
...
 
 
+# TODO(schew2381): Delete this method after the option is removed
+def has_staff_option(user) -> bool:
+    """
+    Checks if the user can access the staff mode for _admin depending on an option.
+    """
+    if (email := getattr(user, "email", None)) is None:
+        return False
+    return email in options.get("staff.user-email-allowlist")
```
Comment (cathteng) [inline src/sentry/auth/staff.py:62]: how will we roll this out to everyone without putting everyone's email on the allowlist?

Comment (schew2381) [reply src/sentry/auth/staff.py:62]: I plan on creating a boolean option and then checking that first in the helper like:
```Python
def has_staff_option(...):
    if options.get("staff.ga-enabled"):
        return True
    ...
```

File: tests/sentry/api/endpoints/test_user_details.py [hunk 2]
```diff
@@ -313,8 +314,8 @@ class UserDetailsStaffUpdateTest(UserDetailsTest):
     method = "put"
 
     @fixture(autouse=True)
-    def use_staff_feature_flag(self):
-        with Feature("auth:enterprise-staff-cookie"):
+    def use_staff_option(self):
+        with override_options({"staff.user-email-allowlist": [self.staff_email]}):
```
Comment (cathteng) [inline tests/sentry/api/endpoints/test_user_details.py:318]: could you remind me what this does again?

File: tests/sentry/api/endpoints/test_user_details.py [hunk 3]
```diff
@@ -57,11 +58,11 @@ def test_superuser_simple(self):
         assert "identities" in resp.data
         assert len(resp.data["identities"]) == 0
 
-    @with_feature("auth:enterprise-staff-cookie")
     def test_staff_simple(self):
         self.login_as(user=self.staff_user, staff=True)
 
-        resp = self.get_success_response(self.user.id)
+        with override_options({"staff.user-email-allowlist": [self.staff_user.email]}):
```
Comment (cathteng) [inline tests/sentry/api/endpoints/test_user_details.py:64]: if you put staff email as a class variable, could you use the `@override_options` decorator?

**Review by cathteng — APPROVED** [review]
it's time 🔓

File: src/sentry/testutils/cases.py [hunk 4]
```diff
@@ -646,6 +646,9 @@ def detect_performance_problems_interceptor(
                     perf_problem.fingerprint = fingerprint
             return perf_problems
 
+        # NOTE: The pre-commit hook fails to parse this line correctly. To
+        # commit changes anyway, ensure all other pre-commit hooks pass then run
+        # `git commit --no-verify`.
```
[the remark this answers is not in the document]
Comment (schew2381) [reply src/sentry/testutils/cases.py:651]: I kinda do cuz my editor keeps formatting this line, and then the pre-commit hook fails 😭

File: tests/sentry/api/endpoints/test_user_details.py (hunk 3 above)
Comment (schew2381) [reply tests/sentry/api/endpoints/test_user_details.py:64]: you know that that's pretty smart, I prob should've done that instead of all the `with` stuff

Comment (schew2381) [reply tests/sentry/api/endpoints/test_user_details.py:64]: Will have to switch everything over sad

Comment (schew2381) [reply tests/sentry/api/endpoints/test_user_details.py:64]: ok on second lookover, some of the stuff makes sense to use `with` for the ones with very few tests

File: tests/sentry/api/endpoints/test_user_details.py (hunk 2 above)
Comment (schew2381) [reply tests/sentry/api/endpoints/test_user_details.py:318]: the fixture is like putting a decorator on every test in the testcase

**Pull request closed without merging**